src/documents.c

```diff
@@ -106,16 +106,35 @@ register_document (const char *uri,
   version = xdp_documents_get_version (documents);
 
   if (for_save)
-    ret = xdp_documents_call_add_named_sync (documents,
-                                             g_variant_new_handle (fd_in),
-                                             basename,
-                                             TRUE,
-                                             TRUE,
-                                             fd_list,
-                                             &doc_id,
-                                             NULL,
-                                             NULL,
-                                             error);
+    {
+      if (version >= 3)
+        {
```

Comment:
I don't think there is a version 3 yet:

The flatpak code has:

  xdp_dbus_documents_set_version (XDP_DBUS_DOCUMENTS (dbus_api), 2);

Reply:
There hopefully will be, see https://github.com/flatpak/flatpak/pull/1335.

Reply:
There is now version 3 as the flatpak patch has been merged.